### PR TITLE
Adding some types hints.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
See the PR https://github.com/keras-team/keras-tuner/pull/163 . If we don't need to support python 2, this should be fine to do it. The types are already displayed in the function signature in the docs, but I plan to also add them in the text description automatically with keras-autodoc. Here is what it looks like right now:
docs_autokeras

### CODE REVIEW

**Comment by jhfjhfj1** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/jhfjhfj1/autokeras/pullRequest?prid=4678707&bid=15652099)

**Comment by jhfjhfj1** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/jhfjhfj1/autokeras/pullRequest?prid=4678707&bid=15652099)

**Comment by haifeng-jin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/jhfjhfj1/autokeras/pullRequest?prid=4678707&bid=15652099)

**Comment by gabrieldemarmiesse** [comment]
@haifeng-jin I believe that adding type hints is a safe move. It doesn't have any runtime consequences. From the pep:

> While these annotations are available at runtime through the usual `__annotations__` attribute, no type checking happens at runtime.

Adding some tool on top of it might. But this PR is focused only on type hints.

I believe it is safe to merge an doesn't necessitate a review from @fchollet or @omalleyt12 .

**Comment by haifeng-jin** [comment]
@gabrieldemarmiesse 
I think it is a good idea to do so if it can save the manual type check and error raising process. 
However, since I am really not familiar with this mechanism there might be some cons we are not aware of. Adding it to all APIs would have some workload.
I would rather wait for the reviews from @fchollet and @omalleyt12 before we proceed.
Thank you for your patience.

**Comment by gabrieldemarmiesse** [comment]
Ok, it doesn't hurt to have another opinion :)

**Comment by gabrieldemarmiesse** [comment]
Also if we decide to type the API surface and do runtime type checks, I'd be happy to implement it :)

File: autokeras/task.py [hunk 1]
```diff
@@ -37,16 +39,16 @@ class ImageClassifier(SupervisedImagePipeline):
...
-                 name='image_classifier',
-                 max_trials=100,
-                 directory=None,
-                 objective='val_loss',
-                 overwrite=True,
-                 seed=None):
+                 num_classes: Optional[int] = None,
+                 multi_label: bool = False,
+                 loss: Union[str, Callable, None] = None,
+                 metrics: Union[str, Callable, None] = None,
```
Comment (fchollet) [inline autokeras/task.py]: This is actually a list.

**Comment by fchollet** [comment]
> However, since I am really not familiar with this mechanism there might be some cons we are not aware of. 

What are some potential issues?

**Comment by haifeng-jin** [comment]
> > However, since I am really not familiar with this mechanism there might be some cons we are not aware of.
> 
> What are some potential issues?

I don't know.
If you don't see any potential issue, then we can just do it.

**Pull request merged**